### ISSUE TITLE
Big problem with Grid WIDTH

### ISSUE THREAD

**Issue by jaimestuardo** [atatanasov/gijgo#434]
I have downloaded GIJGO grid and started testing it. After a little testing, I was wondering if you have seriously consider the width of the grid.

For example, I want to tell you some aspects of the grid that are obvious to me:

When the grid defines a Width, that should be the maximum width. Furthermore, if column widths sum is bigger than grid width, horizontal scroll should appear. At least, that is what the logic says and all grids work that way.

Well.. this grid does not behave that way. No horizontal scroll bar appears and the grid width expands beyond the screen. That behaviour make this grid useless. I recomment this for future enhancements.

**Comment by jaimestuardo** [comment]
You are totally wrong or maybe you did not understand the issue or you are not clear about what responsiveness is.

It is impossible to have a grid with total width equal to the sum of all column widths. I give you a very little example. Suppose you want to see a grid with 10 columns in a mobile phone. Luckily, there will be at most 4 columns visible in the grid. In this case, the grid width will be the width of the display area, but since the grid will contain more columns, it is obvious a scroll should appear in order to move right through the columns using the finger.

In the case of the desktop, a grid with several columns. If the grid width is equal to the sum of columns widths, the grid goes beyond the screen, looking very ugly.

So, this latter case is similar to the case I explained with the mobile phone. Grid width should be the width of the screen, but a scroll should appear to move though the data to see all other columns.

A container DIV solution with scroll is not a real solution, because the scroll will include the paging also, which is not useful (besides, it will look ugly too).

I thought it should be obvious to you since all other grids have such feature and even, from user point of view, this should be something the grid should have.

Please, see this URL: https://datatables.net/examples/basic_init/scroll_x.html. That behaviour is what I was talking abount, not supported by your grid.

Never mind, I did not wait for your reply. I have already removed GIJGO grid from my project and replaced it in all pages with DataTable plugin.

if you are a serious programmer, you should consider having this feature.

Regards
Jaime

**Comment by tomfrey2** [comment]
Hi,
I have to agree with Jaimes. I too have a very wide grid that does not fit on the screen. I have set the width to total of the column widths but that is 1500px. I tried the container div solution but the bottom footer that shows "Displaying Records" is not seen since it is all the way on the right side. it does not look good. Other than that I like the grid ... is it still off the table to allow for the user to set the grid to a certain width and if the actual width is greater create a horizontal scroll bar?  In th emean time I will take a look at the datatables option as suggested.

thanks Tom
